restcomm/restcomm.interpreter/src/test/java/org/restcomm/connect/interpreter/GatherSpeechTest.java

```diff
@@ -0,0 +1,400 @@
+        builder.setCallManager(callManager);
+        builder.setAccount(new Sid("ACae6e420f425248d6a26948c17a9e2acf"));
+        builder.setVersion("2012-04-24");
+        builder.setUrl(requestUri);
+        builder.setMethod("GET");
+        builder.setAsImsUa(false);
+
+        final Props props = new Props(new UntypedActorFactory() {
+            private static final long serialVersionUID = 1L;
+
+            @Override
+            public Actor create() throws Exception {
+                return new VoiceInterpreter(builder.build()) {
+                    @Override
+                    protected ActorRef downloader() {
+                        return observer;
+                    }
+
+                    @Override
+                    protected ActorRef cache(String path, String uri) {
+                        return downloader;
+                    }
+
+                    @Override
+                    public ActorRef getCache() {
+                        return downloader;
+                    }
+                };
+            }
+        });
+        return system.actorOf(props);
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    public void testPartialHangupScenario() throws Exception {
+        new JavaTestKit(system) {
+            {
+                final ActorRef observer = getRef();
+                final ActorRef interpreter = createVoiceInterpreter(observer);
+                interpreter.tell(new StartInterpreter(observer), observer);
+
+                expectMsgClass(GetCallInfo.class);
+                interpreter.tell(new CallResponse(new CallInfo(
+                        new Sid("ACae6e420f425248d6a26948c17a9e2acf"),
+                        CallStateChanged.State.IN_PROGRESS,
+                        CreateCallType.SIP,
+                        "inbound",
+                        new DateTime(),
+                        null,
+                        "test", "test",
+                        "testTo",
+                        null,
+                        null,
+                        false,
+                        false,
+                        false,
+                        new DateTime())), observer);
+
+                expectMsgClass(Observe.class);
+
+                //wait for rcml downloading
+                HttpRequestDescriptor callback = expectMsgClass(HttpRequestDescriptor.class);
+                assertEquals(callback.getUri(), requestUri);
+
+                interpreter.tell(new DownloaderResponse(getOkRcml(requestUri, gatherRcml)), observer);
+
+                expectMsgClass(Collect.class);
+
+                //generate partial response1
+                interpreter.tell(new MediaGroupResponse(new CollectedResult("1", true, true)), observer);
+
+                callback = expectMsgClass(HttpRequestDescriptor.class);
+                assertEquals(callback.getUri(), partialCallbackUri);
+                assertEquals(findParam(callback.getParameters(), "UnstableSpeechResult").getValue(), "1");
```

Comment:
Name of this parameter *UnstableSpeechResult* is confusing. we should make this *PartialSpeechResult*.